File: fizz-buzz/fizz-buzz-hs/fizz-buzz.hs
isMultipleOf3 :: Integer -> Bool
isMultipleOf3 n = n `mod` 3 == 0

isMultipleOf5 :: Integer -> Bool
isMultipleOf5 n = n `mod` 5 == 0

fizzBuzzLoop :: Integer -> IO()
fizzBuzzLoop 101 = putStrLn "End"
fizzBuzzLoop n = do
    if isMultipleOf5 n && isMultipleOf3 n then putStrLn "FizzBuzz"
    else if isMultipleOf3 n then putStrLn "Fizz"
    else if isMultipleOf5 n then putStrLn "Buzz"
    else putStrLn (show n)
    
    fizzBuzzLoop (n + 1)

main :: IO ()
main =  do
    fizzBuzzLoop 1

File: fizz-buzz/fizz-buzz.lua
function is_multiple_of_3(number)
    return number % 3 == 0
end

function is_multiple_of_5(number)
    return number % 5 == 0
end 

function fizz_buzz()
    for counter=1,100 do 
        if is_multiple_of_3(counter) and is_multiple_of_5(counter) then
            print("FizzBuzz")
        elseif is_multiple_of_3(counter) then
            print("Fizz")
        elseif is_multiple_of_5(counter) then
            print("Buzz")
        else
            print(counter)
        end
    end
end

fizz_buzz()